### PR TITLE
add typescript to the project

### PR DESCRIPTION
**Pull request by forybogdanov**
change to typescript

### CODE REVIEW

**Review by preslavnolimit — CHANGES_REQUESTED** [review]

File: src/components/Form.tsx [hunk 1]
```diff
@@ -12,14 +15,15 @@ const Form = ({addCampaign, closeForm}) => {
 
     function getCurrentDate() {
         let today = new Date()
-        let month = today.getMonth()+1
-        if (month < 10) { 
-            month = '0' + month
-        }
+        let month: string
+        let monthNum: number = today.getMonth() + 1
+        monthNum < 10 ? month = '0' + monthNum + '' : month = monthNum + ''
```
[the remark this answers is not in the document]
Comment (forybogdanov) [reply src/components/Form.tsx:20]: Done

File: src/components/Form.tsx [hunk 2]
```diff
@@ -12,14 +15,15 @@ const Form = ({addCampaign, closeForm}) => {
 
     function getCurrentDate() {
         let today = new Date()
-        let month = today.getMonth()+1
-        if (month < 10) { 
-            month = '0' + month
-        }
+        let month: string
+        let monthNum: number = today.getMonth() + 1
```
[the remark this answers is not in the document]
Comment (forybogdanov) [reply src/components/Form.tsx:19]: Done

File: src/components/Form.tsx [hunk 3]
```diff
@@ -12,14 +15,15 @@ const Form = ({addCampaign, closeForm}) => {
...
+        monthNum < 10 ? month = '0' + monthNum + '' : month = monthNum + ''
+        
         return today.getFullYear() + '-' + month + '-' + today.getDate()
     }
-
-    const onSubmit = (e) => {
+    console.log(getCurrentDate());
```
[the remark this answers is not in the document]
Comment (forybogdanov) [reply src/components/Form.tsx:24]: Done

File: src/components/Form.tsx [hunk 4]
```diff
@@ -29,7 +33,7 @@ const Form = ({addCampaign, closeForm}) => {
             alert('No game selected');
             return
         }
-        if (game == 'Other' && !otherGame) {
+        if (game === 'Other' && !otherGame) {
```
[the remark this answers is not in the document]
Comment (forybogdanov) [reply src/components/Form.tsx:36]: Done

File: src/components/Form.tsx [hunk 5]
```diff
@@ -1,5 +1,8 @@
-import { useState } from "react";
-const Form = ({addCampaign, closeForm}) => {
+import React, { MouseEventHandler, useState } from "react";
+import { IFormProps } from "../types";
+const Form = (props: IFormProps) => {
+    const addCampaign: Function = props.addCampaign; 
+    const closeForm: MouseEventHandler = props.closeForm;
```
[the remark this answers is not in the document]
Comment (forybogdanov) [reply src/components/Form.tsx:5]: Done

File: src/components/Form.tsx [hunk 6]
```diff
@@ -63,11 +67,11 @@ const Form = ({addCampaign, closeForm}) => {
             return
         }
         let gameName = game;
-        if (game == 'Other') {
+        if (game === 'Other') {
             gameName = otherGame
         }
         addCampaign({name, description, gameName, startDate, endDate, budget, paidWithCrypto, language})
-        closeForm()
+        closeForm(e) // closeForm requires an argument because its type is mouseEventHandler but I didn't know what to put as an argument
```
[the remark this answers is not in the document]
Comment (forybogdanov) [reply src/components/Form.tsx:74]: Done

File: src/components/Form.tsx [hunk 7]
```diff
@@ -113,11 +117,11 @@ const Form = ({addCampaign, closeForm}) => {
                 </div>
                 <div className="form-control">
                     <label>Budget: </label>
-                    <input type='number' min={0} max={10000} value={budget} onChange={(e) => setBudget(e.target.value)}></input>
+                    <input type='number' min={0} max={10000} value={budget} onChange={(e) => setBudget(parseInt(e.target.value))}></input>
                 </div>
                 <div className="form-control">
                     <label>Can be paid with Crypto currency: </label>
-                    <input type='checkbox' value={paidWithCrypto} checked={paidWithCrypto} onChange={(e) => setPaidWithCrypto(e.currentTarget.checked)}></input>
+                    <input type='checkbox' value={paidWithCrypto ? 1 : 0} checked={paidWithCrypto} onChange={(e) => setPaidWithCrypto(e.currentTarget.checked)}></input>
```
[the remark this answers is not in the document]
Comment (forybogdanov) [reply src/components/Form.tsx:124]: Done

File: src/components/Campaigns.tsx [hunk 8]
```diff
@@ -1,9 +1,12 @@
...
 import Campaign from "./Campaign";
-const Campaigns = ({campaigns}) => {
+
+const Campaigns = (props: ICampaignsProps) => {
     return ( 
         <div className="campaigns-holder">
             <ul>        
-                {campaigns.map((campaign) => (
+                {props.campaigns.map((campaign: ICampaign) => (
                     <li key={campaign.id}><Campaign  campaign={campaign}/></li>
```
[the remark this answers is not in the document]
Comment (forybogdanov) [reply src/components/Campaigns.tsx:10]: Done

**Review by preslavnolimit — APPROVED** [review]

**Pull request merged**